core/domain/fs_services_test.py

```diff
@@ -184,6 +184,31 @@ def test_compress_image_on_prod_mode_with_small_image_size(self):
                     micro_image_content),
                 (22, 22))
 
+    def test_save_original_and_compressed_versions_of_svg_image(self):
+        with python_utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'test_svg.svg'), mode='rb',
+            encoding=None) as f:
+            original_image_content = f.read()
+
+        with self.swap(constants, 'DEV_MODE', False):
+            fs = fs_domain.AbstractFileSystem(
+                fs_domain.GcsFileSystem(
+                    feconf.ENTITY_TYPE_EXPLORATION, self.EXPLORATION_ID))
+
+            self.assertFalse(fs.isfile('image/%s' % self.FILENAME))
+            self.assertFalse(
+                fs.isfile('image/%s' % self.COMPRESSED_IMAGE_FILENAME))
+            self.assertFalse(fs.isfile('image/%s' % self.MICRO_IMAGE_FILENAME))
+
+            fs_services.save_original_and_compressed_versions_of_image(
+                self.FILENAME, 'exploration', self.EXPLORATION_ID,
+                original_image_content, 'image', False)
+
+            self.assertTrue(fs.isfile('image/%s' % self.FILENAME))
+            self.assertTrue(
+                fs.isfile('image/%s' % self.COMPRESSED_IMAGE_FILENAME))
+            self.assertTrue(fs.isfile('image/%s' % self.MICRO_IMAGE_FILENAME))
```

Comment:
Maybe check that the file contents for these are the same.

Reply:
Done.